src/app/donation-start/donation-start-form/donation-start-form-parent.component.ts

```diff
@@ -1652,7 +1652,10 @@ export class DonationStartFormParentComponent implements AfterContentChecked, Af
       if (this.creditPenceToUse === 0) {
         this.amountsGroup.controls.tipAmount!.setValidators([
           Validators.required,
-          Validators.pattern('^[£$]?[0-9]+?(\\.[0-9]{2})?$'),
+          // We allow spaces at start and end of amount inputs because people can easily paste them in
+          // by mistake, and they don't do any harm. Maxlength in the HTML makes sure there can't be so much as
```

Comment:
Not sure we can be totally confident they're harmless without trimming?

There are tests that need updating relating to the validation here.

Reply:
Right. I've updated the tests and added a call to `.trim()` before sanitiseCurrency.